keras_nlp/models/bart/bart_preprocessor.py

```diff
@@ -248,11 +249,44 @@ def presets(cls):
         return copy.deepcopy(backbone_presets)
 
     @classmethod
+    @format_docstring(
+        preprocessor_name="BartPreprocessor",
+        example_preset_name=next(iter(backbone_presets), ""),
+        preset_names='", "'.join(backbone_presets),
+    )
     def from_preset(
         cls,
         preset,
         **kwargs,
     ):
+        """Instantiate {{preprocessor_name}} from preset architecture.
```

Comment:
This is the most interesting change in the lot.

Our `from_preset` docstring for predecessors assumes a single sequence length and single input string. Potentially we could still make that work for seq2seq preprocessors (just take the input string as encoder text, prefix a single start token for decoder text). But it would not work today for sure.

We could also try to adapt the base class, so it makes less assumptions.

Anyway, 0.6 problems to think about. This just makes all our docstring tests pass.

Reply:
I think the right fix is to keep simplifying our `from_preset` docstring. We already have unbelievably long and hard-to-maintain docstrings and repeating material beyond how to load a preset in `from_preset` seems unnecessary.

Reply:
Yeah, sounds good. I can just do that now. Cut out all input data, just show loading.